### ISSUE TITLE
why nan occurs in ME.SparseTensor ?

### ISSUE THREAD

**Issue by zhulf0804** [NVIDIA/MinkowskiEngine#273]
```
batch_coords = ME.utils.batched_coordinates(batch_coords).cuda()
print('batch_feats', batch_feats)
input = ME.SparseTensor(torch.cat(batch_feats, dim=0).float(),
                        coordinates=batch_coords,
                        quantization_mode=ME.SparseTensorQuantizationMode.UNWEIGHTED_AVERAGE)
print('input size: ', input.coordinates.size())
print('input', input)
```

The result is as follows, `nan` in `input. features`, who can help me solve the problem?

batch_feats [tensor([[-0.7738, -0.2920, -0.5622],
        [-0.2925, -0.3223, -0.9003],
        [ 0.8590,  0.5119,  0.0000],
        ...,
        [ 0.7805, -0.2634,  0.5670],
        [-0.3057, -0.1456,  0.9409],
        [-0.9959,  0.0899,  0.0000]], device='cuda:0')]
input size:  torch.Size([768, 4])
input SparseTensor(
  coordinates=tensor([[  0,   7,   3,  13],
        [  0,   2,  -6, -13],
        [  0, -15,  -1,  -5],
        ...,
        [  0,   9,  -5,   7],
        [  0,  -4,  -2,  12],
        [  0, -13,   6,   4]], device='cuda:0', dtype=torch.int32)
  features=tensor([[ 0.0239,  0.0238, -0.0336],
        [    nan,     nan,     nan],
        [    nan,     nan,     nan],
        ...,
        [    nan,     nan,     nan],
        [    nan,     nan,     nan],
        [    nan,     nan,     nan]], device='cuda:0')
  coordinate_map_key=coordinate map key:[1, 1, 1]
  coordinate_manager=CoordinateMapManagerGPU_c10(
	[1, 1, 1]:	CoordinateMapGPU:768x4
	algorithm=MinkowskiAlgorithm.DEFAULT
  )
  spatial dimension=3)

**Comment by zhulf0804** [comment]
Maybe some bugs in the `ME.SparseTensorQuantizationMode.UNWEIGHTED_AVERAGE` implementation.

I solved the problem by replacing `ME.SparseTensorQuantizationMode.UNWEIGHTED_AVERAGE` with `ME.SparseTensorQuantizationMode.RANDOM_SUBSAMPLE` while maintaining the accuracy.

Moreover, I modified [https://github.com/NVIDIA/MinkowskiEngine/blob/v0.5/MinkowskiEngine/MinkowskiTensorField.py#L65](https://github.com/NVIDIA/MinkowskiEngine/blob/v0.5/MinkowskiEngine/MinkowskiTensorField.py#L65) as follows:

```
assert quantization_mode in [
            SparseTensorQuantizationMode.UNWEIGHTED_AVERAGE,
            SparseTensorQuantizationMode.UNWEIGHTED_SUM,
            SparseTensorQuantizationMode.RANDOM_SUBSAMPLE,
        ], "invalid quantization mode"
```

**Comment by zhulf0804** [comment]
Solved.

**Comment by chrischoy** [comment]
Could you please share the MinkowskiEngine commit id, cuda version, and driver version?

**Comment by zhulf0804** [comment]
Hi @chrischoy ,
Glad to hear from you.

branch: v0.5
commit id: commit 007a4ca76afb5d4d4f9fe8689ae76e326bbe75a7
cuda: V10.1.243
driver: 430.34

**Comment by chrischoy** [comment]
I can't replicate this error on CUDA 10.1.243.
Can you please post an example code that replicates this error?
